sympy/solvers/tests/test_solveset.py

```diff
@@ -177,6 +177,9 @@ def test_domain_check():
     assert domain_check(x, x, oo) is False
     assert domain_check(0, x, oo) is False
 
+def test_issue_11536():
+    assert solveset(0**x - 100, x, S.Reals) == ConditionSet(x, Eq(0**x - 100, 0), S.Reals)
+    assert solveset(0**x - 1, x, S.Reals) == FiniteSet(0)
```

Comment:
No, it covers only points 1 and 2.
For point 3, there was [this](https://github.com/sympy/sympy/blob/master/sympy/solvers/tests/test_solveset.py#L277) testcase in `solveset` itself that would raise an exception, so I thought not to write a testcase.
For point 4 and 5 I can add tests.